models.py

```diff
@@ -40,15 +41,42 @@
                        }
 
 
-class PyTorchUNet(Model):
-    def __init__(self, architecture_config, training_config, callbacks_config):
+class BasePyTorchUNet(Model):
+    def __init__(self, architecture_config, training_config, callbacks_config, validation_pipeline):
         super().__init__(architecture_config, training_config, callbacks_config)
         self.set_model()
         self.weight_regularization = weight_regularization_unet
         self.optimizer = optim.Adam(self.weight_regularization(self.model, **architecture_config['regularizer_params']),
                                     **architecture_config['optimizer_params'])
-        self.loss_function = [('multichannel_map', multiclass_segmentation_loss, 1.0)]
+        self.loss_function = None
         self.callbacks = callbacks_unet(self.callbacks_config)
+        self.validation_pipeline = validation_pipeline
+
+    def fit(self, datagen, validation_datagen=None, meta_valid=None, *args, **kwargs):
+        self._initialize_model_weights()
+
+        self.model = nn.DataParallel(self.model)
+
+        if torch.cuda.is_available():
+            self.model = self.model.cuda()
+
+        self.callbacks.set_params(self, validation_datagen=validation_datagen, meta_valid=meta_valid)
```

Comment:
@taraspiotr This shows that we can pass meta_valid to callback constructor rather than refactor Unet class